test/gallery-route-test.js

```diff
@@ -0,0 +1,241 @@
+        this.tempToken = token;
+        debug('token', this.tempToken);
+        done();
+      })
+      .catch(done);
+    });
+
+    it('should return a gallery', done => {
+      debug('inside it block', this.tempToken);
+      request.post(`${url}/api/gallery`)
+      .send(exampleGallery)
+      .set({
+        Authorization: `Bearer ${this.tempToken}`
+      })
+      .end((err, res) => {
+        if (err) return done(err);
+        let date = new Date(res.body.created).toString();
+        expect(res.body.name).to.equal(exampleGallery.name);
+        expect(res.body.desc).to.equal(exampleGallery.desc);
+        expect(res.body.userID).to.equal(this.tempUser._id.toString());
+        expect(date).to.not.equal('Invalid Date');
+        done();
+      });
+    });
+
+    it('should return a 401 error with no token included', done => {
+      debug('inside POST 401 error test');
+      request.post(`${url}/api/gallery`)
+      .send(exampleGallery)
+      .end(err => {
+        expect(err.status).to.equal(401);
+        done();
+      });
+    });
+
+    it('should return a 400 error with no/invalid body', done => {
+      debug('inside POST 400 error test');
+      request.post(`{url}/api/gallery`)
+      .send()
+      .end(err => {
+        expect(err.status).to.equal(400);
+        done();
+      });
+    });
+  });
+
+  describe('GET: /api/gallery/:id', () => {
+    before( done => {
+      new User(exampleUser)
+      .generatePasswordHash(exampleUser.password)
+      .then( user => user.save())
+      .then( user => {
+        this.tempUser = user;
+        return user.generateToken();
+      })
+      .then( token => {
+        this.tempToken = token;
+        done();
+      })
+      .catch(done);
+    });
+
+    before( done => {
+      exampleGallery.userID = this.tempUser._id.toString();
+      new Gallery(exampleGallery).save()
+      .then( gallery => {
+        this.tempGallery = gallery;
+        done();
+      })
+      .catch(done);
+    });
+
+    after( () => {
+      delete exampleGallery.userID;
+    });
+
+    it('should return a gallery', done => {
+      request.get(`${url}/api/gallery/${this.tempGallery._id}`)
+      .set({
+        Authorization: `Bearer ${this.tempToken}`
+      })
+      .end((err,res) => {
+        if (err) return done(err);
+        let date = new Date(res.body.created).toString();
+        expect(res.body.name).to.equal(exampleGallery.name);
+        expect(res.body.desc).to.equal(exampleGallery.desc);
+        expect(res.body.userID).to.equal(this.tempUser._id.toString());
+        expect(date).to.not.equal('Invalid Date');
+        done();
+      });
+    });
+
+    it('should return a 401 error with no token included', done => {
+      debug('inside GET 401 error test');
+      request.get(`${url}/api/gallery/${this.tempGallery._id}`)
+      .end(err => {
+        expect(err.status).to.equal(401);
+        done();
+      });
+    });
+
+    describe('GET: api/gallery', done => {
```

Comment:
don't use done in describe block, change it to :
`describe('GET: api/gallery', () => {..}`